Apps/Src/Task_BatteryBalance.c

```diff
@@ -1,15 +1,17 @@
 #include "os.h"
 #include "Tasks.h"
 #include <stdlib.h>
+#include "BatteryBalancing.h"
+
+extern cell_asic Minions[NUM_MINIONS];
 
 void Task_BatteryBalance(void *p_arg) {
     (void)p_arg;
 
     OS_ERR err;
 
-    while(1) {
-        // Todo: Battery balancing
-
+    while(1){
+        Balancing_Balance(Minions);
```

Comment:
If you access the Minions array, you need to use the `MinionsASIC_Mutex` (see Voltage.c). I think it would probably be best to add the mutex calls to the functions in Balancing.c that access this array, since if you add it on the task level, this will stop the voltage and temperature thread from running most of the time